File: BackEnd/Zalo_mini/Zalo_mini/Models/call.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class call
    {
        public call()
        {
            call_participants = new HashSet<call_participant>();
        }

        public long id { get; set; }
        public long conversation_id { get; set; }
        public long caller_id { get; set; }
        public string call_type { get; set; } = null!;
        public string? status { get; set; }
        public DateTime? started_at { get; set; }
        public DateTime? answered_at { get; set; }
        public DateTime? ended_at { get; set; }
        /// <summary>
        /// Duration in seconds
        /// </summary>
        public int? duration { get; set; }

        public virtual user caller { get; set; } = null!;
        public virtual conversation conversation { get; set; } = null!;
        public virtual ICollection<call_participant> call_participants { get; set; }
    }
}

File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/Call.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "Call")
@Table(name = "calls")
public class Call {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "conversation_id", nullable = false)
    private Conversation conversation;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "caller_id", nullable = false)
    private User caller;

    @Lob
    @Column(name = "call_type", nullable = false)
    private String callType;

    @ColumnDefault("'initiated'")
    @Lob
    @Column(name = "status")
    private String status;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "started_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date startedAt;

    @Column(name = "answered_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date answeredAt;

    @Column(name = "ended_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date endedAt;

    @Column(name = "duration")
    private Integer duration;

    @OneToMany(mappedBy = "call", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<CallParticipant> participants;

}